### ISSUE TITLE
Bug in the construction of a gdual constant via gdual x(1., "x", 0) or gdual x("x", 0);

### ISSUE THREAD

**Issue by darioizzo** [darioizzo/audi#18]
@neXyon noted that:

If we use the following expression to construct a gdual constant (order 0):
```c++
 gdual x(1., "x", 0);
 std::cout << x << '\n';
```
we internally build a piranha polynomial 1. + dx (see https://github.com/darioizzo/audi/blob/master/src/gdual.hpp#L300) which contains an order 1 term. This is an evident bug when we stream out the gdual, causes the assertion upon destruction to fail and has  consequences in computations involving x.

**Comment by darioizzo** [comment]
The problem arises in the two constructors (L250, L300) as they construct, in the initializer list, the 
piranha polynomial via its constructor (https://github.com/bluescarni/piranha/blob/master/src/polynomial.hpp#L1218) that "construct a univariate polynomial made of a single term with unitary coefficient and exponent", ignoring the corner case of having requested a 0 order.

To fix this:
 1 Throw an error in the audi constructors if the order requested is zero, thus forcing the user to use the other constructor, or
 2 check in the constructor body for this corner case

Suggested solution (2) add in both constructors:
```c++
if (order==0) {
 m_p- = std::string("d") + symbol
}
```
which should leave dx in the symbol set, but remove the coefficient 1 from dx. @bluescarni is this a good solution?

**Comment by darioizzo** [comment]
closed by commit c30b47b42d4808336ef5a4e497db3c908f66b925
